.github/workflows/e2e.yml

```diff
@@ -0,0 +1,233 @@
+name: E2E
+
+on:
+  workflow_dispatch:
+  schedule:
+    # Run everyday day at 9:00 AM
+    - cron: '0 9 * * *'
+  pull_request:
+    paths-ignore:
+      - 'scripts/**'
+      - '*.md'
+  push:
+    branches:
+      - 'release/*"'
+      - 'main'
+      - 'master'
+
+env:
+  GOARCH: amd64
+  CGO_ENABLED: 0
+  SETUP_GO_VERSION: '^1.19'
+  HOST_NAME: telemetry-test  # for `start` script
+  CLUSTER_NAME: telemetry-test
+  TOKEN: ${{ secrets.TOKEN }}
+  REPO: telemetry-stats
+  REPO_URL: $GITHUB_SERVER_URL/$GITHUB_REPOSITORY
+  REPOSITORY: ${{ github.repository }}
+  BRANCH: ${{ github.ref_name }}
+  TELEMETRY_REPO: https://github.com/rancher/telemetry
+  TELEMETRY_BRANCH: ${{ vars.TELEMETRY_BRANCH || 'release' }}
+
+jobs:
+  single-cluster:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          path: telemetry-stats
+
+      - name: Get latest tag for telemetry
+        uses: actions/github-script@v6
+        id: latest_tag
+        if: ${{ env.TELEMETRY_BRANCH == 'release' }}
+        with:
+          script: |
+            const { data } = await github.rest.repos.getLatestRelease({
+              owner: 'rancher',
+              repo: 'telemetry'
+            })
+            return data.tag_name;
+
+      - name: Replace TELEMETRY_BRANCH with latest tag
+        if: ${{ env.TELEMETRY_BRANCH == 'release' }}
+        run: |
+          export ACTUAL_TELEMETRY_BRANCH=${{ steps.latest_tag.outputs.result }}
+          echo "ACTUAL_TELEMETRY_BRANCH=${ACTUAL_TELEMETRY_BRANCH}" >> $GITHUB_ENV
+
+      - name: Set ACTUAL_TELEMETRY_BRANCH to the value of TELEMETRY_BRANCH if still unset
+        if: ${{ env.TELEMETRY_BRANCH != 'release' }}
+        run: |
+          echo "ACTUAL_TELEMETRY_BRANCH=${TELEMETRY_BRANCH}" >> $GITHUB_ENV
+
+      - name: Print TELEMETRY_BRANCH and ACTUAL_TELEMETRY_BRANCH
+        run: |
+          echo "TELEMETRY_BRANCH=${TELEMETRY_BRANCH}"
+          echo "ACTUAL_TELEMETRY_BRANCH=${ACTUAL_TELEMETRY_BRANCH}"
+
+      - uses: actions/checkout@v3
+        with: 
+          repository: rancher/telemetry
+          path: telemetry
+          ref: ${{ env.ACTUAL_TELEMETRY_BRANCH }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.SETUP_GO_VERSION }}
+
+      - name: Compile & containerize Telemetry
+        if: ${{ env.TELEMETRY_BRANCH != 'release' }}
+        run: |
+          set -x
+
+          # Compile using dapper
+          cd telemetry
+          curl -sL https://releases.rancher.com/dapper/latest/dapper-$(uname -s)-$(uname -m) > /usr/local/bin/dapper
+          chmod +x /usr/local/bin/dapper
+          dapper
+
+          # Build container image
+          export image_name="rancher/telemetry:${ACTUAL_TELEMETRY_BRANCH}"
+
+          docker build \
+            --network=host \
+            -f package/Dockerfile \
+            -t ${image_name} \
+            .
```

Comment:
I think you're right, it might not be necessary, instead a `docker tag ...` might be sufficient here, provided name and tag of the container image created by `dapper` doesn't change. I'll need to check.

Reply:
Checked. There's little I can do to specify which tag I want it to have. While it mostly should be `dev`, it is not guaranteed to be. Dapper calls `scripts/ci`, which calls `scripts/version`. The latter determines the tag, which is then used in `scripts/package`. By re-containerizing I don't have to fiddle with the CI and don't need to determine which tag has been used, to re-tag afterwards. It might not be ideal, but works. Do you have a strong opinion about me having to improve it?